tests/atomistics/structure/test_neighbors.py

```diff
@@ -223,6 +223,42 @@ def test_get_local_shells(self):
         self.assertTrue(np.array_equal(shells, neigh.get_local_shells(cluster_by_vecs=True)))
         self.assertFalse(np.array_equal(shells, neigh.get_local_shells()))
 
+    def test_get_global_shells_ragged(self):
+        structure = CrystalStructure(
+            elements='Al', lattice_constants=4, bravais_basis='fcc'
+        ).repeat(2)
+        del structure[0]
+        neigh = structure.get_neighbors(cutoff_radius=3.5, num_neighbors=None)
+        self.assertEqual(np.sum(neigh.get_global_shells()==-1), 12)
```

Comment:
Why should it be -1 for 12 neighbors? Shouldn't those be correctly classified?

Reply:
Oh that's probably an important point that should have been better documented in the docstring. -1 is the equivalent of `np.inf` for `distances`. So there are now three conventions that coexist:

- `np.inf` for `distances` and `vecs`
- `n_atoms+1` for `indices`
- -1 for `shells`

I hope we can live with these ones.